File: Codechef/Codechef_NOV_LONG_2020/codechef.02.py
try:
    test = int(input())
    for _ in range(test):
        k = -1 
        n = int(input())
        arr = list(map(int, input().split()))
        res = [0] * n
        
        temp = 4000000

        for i in range(n):
            if arr[i] == i + 1:
                res[i] = temp 
                temp -= 1 

        for i in range(n):
            if arr[i] != (i + 1):
                res[i] = res[arr[i] - 1]

        for ele in res:
            print(ele, end = " ")

        print("")
except:
    pass

File: Codechef/2020/Codechef_AUG_LONG_2020/codechef.02.c
#include<stdio.h>
int main()
{
    int test, q;
    scanf("%d", &test);
    for(q = 0; q < test; q++)
    {
        int i, temp = 0, min = -1, flag = 0, n, k, res = -1;
        scanf("%d %d", &n, &k);

        int arr[n];

        for(i = 0; i< n; i++)
        {
            scanf("%d", &arr[i]);
        }

        for(i = 0; i < n; i++)
        {
            if(arr[i] <= k && k % arr[i] == 0)
            {
                temp = k / arr[i];
                flag++;

                if(flag == 1)
                {
                    min = temp;
                    res = arr[i];
                }

                else 
                {
                    if(temp < min)
                    {
                        min = temp;
                        res = arr[i];
                    }
                }
            }
        }

        if(res != -1)
        {
            printf("%d\n", res);
        }
        else 
        {
            printf("-1\n");
        }
    }
return 0;
}